src/Skipper/Matcher/FileInfoMatcher.php

```diff
@@ -61,6 +61,11 @@ private function doesFileMatchPattern(string $filePath, string $ignoredPath): bo
             return true;
         }
 
+        // realpathMatcher cannot resolve wildcards -> return early to prevent unnecessary IO
+        if (str_contains($ignoredPath, '*')) {
```

Comment:
@staabm I just check that file with `*` in middle name is valid name:

```diff
➜  rector-src git:(main) ✗ bin/rector process rules/Privatization/foo*bar.php
 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
1 file with changes
===================

1) rules/Privatization/foo*bar.php:0

    ---------- begin diff ----------
@@ @@
 <?php

-echo strpos('a', 'a');
+declare(strict_types=1);
+
+echo strpos('a', 'a');
    ----------- end diff -----------
```

do you think it make sense to revert it?

Reply:
Just re-check it and register in skip config, seems fine and likely no body will create a file with `*` in middle name, I will keep on eye on this in case we have this kind of case in the future :)

Reply:
thanks for checking.

I think the command you used is wrong:

`bin/rector process rules/Privatization/foo*bar.php`

if you want to use a path in a shell which contains a `*` sign, you need to use single quotes arround it:
`bin/rector process 'rules/Privatization/foo*bar.php'`

otherwise the shell will expand the `*`

Reply:
the command is working ok, as it works before it checked `*`, which use `realpath`, `foo*bar.php` is skipped, `foo_bar.php` is scanned.

Reply:
that seems unfortunatelly not

Reply:
on my machine, the example you used did not work before and not after the PR:

```
mstaab@NB-COMPLEX-61 MINGW64 /c/dvl/Workspace/rector-src-staabm (main)
$ bin/rector process rules/Privatization/foo*bar.php --dry-run

                                                                                                                        
 [ERROR] The given paths do not match any files                                                                         
                                                                                                                        


mstaab@NB-COMPLEX-61 MINGW64 /c/dvl/Workspace/rector-src-staabm (main)
$ bin/rector process 'rules/Privatization/foo*bar.php' --dry-run

                                                                                                                        
 [ERROR] The given paths do not match any files                                                                         
                                                                                                                  

```